// react-native-paper-dates/lib/commonjs/Date/DateRangeInput.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = DateRangeInput;

var React = _interopRequireWildcard(require("react"));

var _reactNative = require("react-native");

var _reactNativePaper = require("react-native-paper");

var _TextInputMask = _interopRequireDefault(require("../TextInputMask"));

var _DatePickerModal = _interopRequireDefault(require("./DatePickerModal"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function () { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function DateRangeInput() {
  const [visible, setVisible] = React.useState(false);
  const onDismiss = React.useCallback(() => {
    setVisible(false);
  }, [setVisible]);
  const onConfirm = React.useCallback( // eslint-disable-next-line @typescript-eslint/no-unused-vars
  ({
    startDate,
    endDate
  }) => {
    setVisible(false);
  }, [setVisible]);
  return /*#__PURE__*/React.createElement(_reactNative.View, {
    style: {
      flexDirection: 'row',
      alignItems: 'center'
    }
  }, /*#__PURE__*/React.createElement(_reactNative.View, {
    style: {
      flex: 1
    }
  }, /*#__PURE__*/React.createElement(_TextInputMask.default, {
    value: '',
    keyboardType: 'numeric',
    placeholder: 'DD-MM-YYY',
    mask: 'DD-MM-YYY',
    onChangeText: () => {}
  }), /*#__PURE__*/React.createElement(_reactNativePaper.Text, null, "Van")), /*#__PURE__*/React.createElement(_reactNative.View, null, /*#__PURE__*/React.createElement(_reactNativePaper.Text, {
    style: {
      fontSize: 16,
      marginLeft: 12,
      marginRight: 12
    }
  }, "to"), /*#__PURE__*/React.createElement(_reactNativePaper.Text, {
    style: {
      opacity: 0
    },
    accessible: false
  }, "tot")), /*#__PURE__*/React.createElement(_reactNative.View, {
    style: {
      flex: 1
    }
  }, /*#__PURE__*/React.createElement(_TextInputMask.default, {
    value: '',
    keyboardType: 'numeric',
    placeholder: 'DD-MM-YYY',
    mask: 'DD-MM-YYY',
    onChangeText: () => {}
  }), /*#__PURE__*/React.createElement(_reactNativePaper.Text, null, "Tot")), /*#__PURE__*/React.createElement(_reactNative.View, null, /*#__PURE__*/React.createElement(_reactNativePaper.IconButton, {
    icon: "calendar",
    onPress: () => setVisible(true)
  }), /*#__PURE__*/React.createElement(_reactNativePaper.Text, {
    style: {
      opacity: 0
    },
    accessible: false
  }, "tot")), /*#__PURE__*/React.createElement(_DatePickerModal.default, {
    mode: "range",
    visible: visible,
    onDismiss: onDismiss,
    onConfirm: onConfirm,
    startDate: undefined,
    endDate: undefined
  }));
}
