// react-native-paper-dates/lib/module/Date/DateRangeInput.js
import * as React from 'react';
import { View } from 'react-native';
import { Text, IconButton } from 'react-native-paper';
import TextInputWithMask from '../TextInputMask';
import DatePickerModal from './DatePickerModal';
export default function DateRangeInput() {
  const [visible, setVisible] = React.useState(false);
  const onDismiss = React.useCallback(() => {
    setVisible(false);
  }, [setVisible]);
  const onConfirm = React.useCallback( // eslint-disable-next-line @typescript-eslint/no-unused-vars
  ({
    startDate,
    endDate
  }) => {
    setVisible(false);
  }, [setVisible]);
  return /*#__PURE__*/React.createElement(View, {
    style: {
      flexDirection: 'row',
      alignItems: 'center'
    }
  }, /*#__PURE__*/React.createElement(View, {
    style: {
      flex: 1
    }
  }, /*#__PURE__*/React.createElement(TextInputWithMask, {
    value: '',
    keyboardType: 'numeric',
    placeholder: 'DD-MM-YYY',
    mask: 'DD-MM-YYY',
    onChangeText: () => {}
  }), /*#__PURE__*/React.createElement(Text, null, "Van")), /*#__PURE__*/React.createElement(View, null, /*#__PURE__*/React.createElement(Text, {
    style: {
      fontSize: 16,
      marginLeft: 12,
      marginRight: 12
    }
  }, "to"), /*#__PURE__*/React.createElement(Text, {
    style: {
      opacity: 0
    },
    accessible: false
  }, "tot")), /*#__PURE__*/React.createElement(View, {
    style: {
      flex: 1
    }
  }, /*#__PURE__*/React.createElement(TextInputWithMask, {
    value: '',
    keyboardType: 'numeric',
    placeholder: 'DD-MM-YYY',
    mask: 'DD-MM-YYY',
    onChangeText: () => {}
  }), /*#__PURE__*/React.createElement(Text, null, "Tot")), /*#__PURE__*/React.createElement(View, null, /*#__PURE__*/React.createElement(IconButton, {
    icon: "calendar",
    onPress: () => setVisible(true)
  }), /*#__PURE__*/React.createElement(Text, {
    style: {
      opacity: 0
    },
    accessible: false
  }, "tot")), /*#__PURE__*/React.createElement(DatePickerModal, {
    mode: "range",
    visible: visible,
    onDismiss: onDismiss,
    onConfirm: onConfirm,
    startDate: undefined,
    endDate: undefined
  }));
}
